src/repositories/api_key/api_key_redis.rs

```diff
@@ -0,0 +1,547 @@
+        key_prefix: String,
+    ) -> Result<Self, RepositoryError> {
+        if key_prefix.is_empty() {
+            return Err(RepositoryError::InvalidData(
+                "Redis key prefix cannot be empty".to_string(),
+            ));
+        }
+
+        Ok(Self {
+            client: connection_manager,
+            key_prefix,
+        })
+    }
+
+    /// Generate key for api key data: apikey:{api_key_id}
+    fn api_key_key(&self, api_key_id: &str) -> String {
+        format!("{}:{}:{}", self.key_prefix, API_KEY_PREFIX, api_key_id)
+    }
+
+    /// Generate key for api key list: apikey_list (paginated list of api key IDs)
+    fn api_key_list_key(&self) -> String {
+        format!("{}:{}", self.key_prefix, API_KEY_LIST_KEY)
+    }
+
+    async fn get_by_ids(
+        &self,
+        ids: &[String],
+    ) -> Result<BatchRetrievalResult<ApiKeyModel>, RepositoryError> {
+        if ids.is_empty() {
+            debug!("No api key IDs provided for batch fetch");
+            return Ok(BatchRetrievalResult {
+                results: vec![],
+                failed_ids: vec![],
+            });
+        }
+
+        let mut conn = self.client.as_ref().clone();
+        let keys: Vec<String> = ids.iter().map(|id| self.api_key_key(id)).collect();
+
+        let values: Vec<Option<String>> = conn
+            .mget(&keys)
+            .await
+            .map_err(|e| self.map_redis_error(e, "batch_fetch_api_keys"))?;
+
+        let mut apikeys = Vec::new();
+        let mut failed_count = 0;
+        let mut failed_ids = Vec::new();
+        for (i, value) in values.into_iter().enumerate() {
+            match value {
+                Some(json) => match self.deserialize_entity(&json, &ids[i], "apikey") {
+                    Ok(apikey) => apikeys.push(apikey),
+                    Err(e) => {
+                        failed_count += 1;
+                        error!("Failed to deserialize api key {}: {}", ids[i], e);
+                        failed_ids.push(ids[i].clone());
+                    }
+                },
+                None => {
+                    warn!("Plugin {} not found in batch fetch", ids[i]);
+                }
+            }
+        }
+
+        if failed_count > 0 {
+            warn!(
+                "Failed to deserialize {} out of {} api keys in batch",
+                failed_count,
+                ids.len()
+            );
+        }
+
+        Ok(BatchRetrievalResult {
+            results: apikeys,
+            failed_ids,
+        })
+    }
+}
+
+impl fmt::Debug for RedisApiKeyRepository {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "RedisApiKeyRepository {{ key_prefix: {} }}",
+            self.key_prefix
+        )
+    }
+}
+
+#[async_trait]
+impl ApiKeyRepositoryTrait for RedisApiKeyRepository {
+    async fn create(&self, entity: ApiKeyModel) -> Result<ApiKeyModel, RepositoryError> {
```

Comment:
With latest PR https://github.com/OpenZeppelin/openzeppelin-relayer/pull/392/files#diff-c6dc9b6c92493d851c7ff29b7339c31c7b553209c9d753978bde35b80f4f5304

I have added logic for encrypting sensitive values.

You can prepare logic now by introducing repository model and later we can just add serialize and deserialize functions to field with sensitive value.

We should propagate it here once it lands to main.